src/modules/vacancyTypes.ts

```diff
@@ -0,0 +1,17 @@
+import { Group, Vacancy, User } from 'prisma/prisma-client';
+
+import { Nullish } from '../utils/types';
+import { Theme } from '../utils/theme';
+
+export interface UserSettings {
+    userId: string;
+    theme: Theme;
+}
+
+export type VacancyWithGroup = Vacancy & {
+    group: Group;
+};
+
+export type VacancyWithGroupAndUser = VacancyWithGroup & {
```

Comment:
This kind of naming of types was out of control in hire. Let's try do compound types instead, like in userTypes.tsx - like `User & UserSupervisor`. This way every component can only ask for things that it really needs